serializer.h

```diff
@@ -1757,6 +1765,122 @@ void serialize(Archive & archive, const polymorphic_wrapper<Type> & object)
     registry_instance.serialize(archive, *object);
 }
 
+/**
+ * Support types introduced in c++17
+ */
+#if __cplusplus > 201402L
+/**
+ * The base function of the variant serialization helper function.
+ */
+template <typename Archive, typename Variant,
+    typename = typename Archive::loading
+>
+void serialize(Archive &, const Variant &, std::size_t)
+{
+}
+
+/**
+ * Serialize std::variant, operates on loading(input) archives.
+ * Enumerate on all the variant types, until the active type is
+ * reached, and serialize it.
+ */
+template <typename Archive, typename VariantItem, typename... VariantItems,
+    typename Variant, typename = typename Archive::loading
+>
+void serialize(Archive & archive, Variant & variant, std::size_t index)
+{
+    if (0 == index) {
+        VariantItem item;

```

Comment:
I think we need to construct this in a way that a private default constructor works, look at how this is done with "access" structure.

Reply:
Done, not sure if this is what you meant.